src/Mod/PartDesign/App/FeaturePad.cpp

```diff
@@ -115,18 +129,62 @@ App::DocumentObjectExecReturn *Pad::execute(void)
         base = TopoDS_Shape();
     }
 
-
     // get the Sketch plane
-    Base::Placement SketchPos    = obj->Placement.getValue();
-    Base::Vector3d  SketchVector = getProfileNormal();
+    Base::Placement SketchPos = obj->Placement.getValue();
+    // get the normal vector of the sketch
+    Base::Vector3d SketchVector = getProfileNormal();
 
     try {
+        // get user-defined decimals
+        int precision = Base::UnitsApi::getDecimals();
+
         this->positionByPrevious();
         TopLoc_Location invObjLoc = this->getLocation().Inverted();
 
         base.Move(invObjLoc);
 
-        gp_Dir dir(SketchVector.x,SketchVector.y,SketchVector.z);
+        Base::Vector3d paddingDirection;
+
+        // use the given vector if necessary
+        if (!UseCustomVector.getValue()) {
+            paddingDirection.x = SketchVector.x;
+            paddingDirection.y = SketchVector.y;
+            paddingDirection.z = SketchVector.z;
+        }
+        else {
+            // if null vector, use SketchVector
+            if ( (RoundTo(XSkew.getValue(), precision) == 0.0)

```

Comment:
This means result of recompute will be dependent on user preferences. This is not good. Use `Precision::Confusion` instead, please.